Parse drag-dropped folders and subfolders in Chrome

benjamincarp opened the pull request:
This change allows resumable to parse through folders that are dragged and dropped into the drop zone if the browser supports it (just Chrome for now).  It still handles them in pretty much the usual fashion for other browsers.  

In the interest of full disclosure, the general algorithm for this was found in a blog post at http://code.flickr.net/2012/12/10/drag-n-drop/ 

In addition to simply parsing the directories, I also add a property of .relativePath to files found inside a folder and use that to set the resumableFile.relativePath if file.webkitRelativePath is nothing.

steffentchr commented:
Looks great -- but would you be able to add a few more lines of comments to the code? Mainly notes about the method for folder support and about the current Chrome-only status?

benjamincarp commented:
I added several more comments as well as function headers (I can remove these if you think they add too much clutter).

I also did more testing with more complicated sub folder structures and fixed a couple of bugs I found.

steffentchr commented:
Awesome! Merging with thanks!
